### PR TITLE
[new] Add mapbox-gl via single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by jasonbarry**
Pull request for issue: #10654 (supersedes PR #10734, #10745)

# Profile of the lib
 * Git repository (required): https://github.com/mapbox/mapbox-gl-js
 * Official website (optional, not the repository): https://www.mapbox.com/mapbox-gl-js/api/
 * NPM package url (optional): https://www.npmjs.com/package/mapbox-gl
 * License and its reference: [BSD-3-Clause](https://www.npmjs.com/package/mapbox-gl)
 * GitHub / Bitbucket popularity (required): 
   - Count of watchers: 221
   - Count of stars: 1554
   - Count of forks: 367
 * NPM download stats (optional):
   - Downloads in the last day: 962
   - Downloads in the last week: 11352
   - Downloads in the last month: 46516

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@jasonbarry would you like to also make the commit message more clear about the detail? (A little more like the PR title)

**Comment by jasonbarry** [comment]
@PeterDaveHello Any way to do that directly on GitHub? I'd rather not do a sparse checkout since my internet connection at the moment is poor.

**Comment by PeterDaveHello** [comment]
Then no, unless you'd like to send a new PR again 😄

**Comment by jasonbarry** [comment]
@PeterDaveHello no thanks 😋

**Comment by jasonbarry** [comment]
Anything I can do here to expedite review? Thanks in advance!

File: ajax/libs/mapbox-gl/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "map",
+    "mapbox", 
+    "mapping"
+  ],
+  "npmName": "mapbox-gl",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "mapbox-gl.css",
```
Comment (hare1039) [inline ajax/libs/mapbox-gl/package.json]: We need all files under `dist`
Please use
```
"files": [
  "**/*"
]
```

Comment (jasonbarry) [inline ajax/libs/mapbox-gl/package.json]: The other files are just sourcemaps, should I still use "**/*"?

Comment (hare1039) [inline ajax/libs/mapbox-gl/package.json]: Yes please

**Comment by jasonbarry** [comment]
Will make requested changes in subsequent PR.

**Pull request closed without merging**